### DISCUSSION TITLE
Section 04: "PyTorch Custom Datasets": Compare Our Custom Dataset Class to the original ImageFolder Class- TypeError

### DISCUSSION THREAD

**Discussion by marvelousxmax** [mrdbourke/pytorch-deep-learning#939] [Q&A]
Hello! I am having trouble on video "Compare Our Custom Dataset Class to the original ImageFolder Class" (Video point 03:22). I am receiving this TypeError message after writing the code: 

![image](https://github.com/mrdbourke/pytorch-deep-learning/assets/131393386/587b551e-af1f-499e-bba4-303b52a9de36)

I have rewritten the Custom Dataset Class from scratch and still receiving the error. 

Any explanation would be helpful. Thank you.

**Comment by sametucuncu** [comment]
How did you create the imagefoldercustom class? Did you use Dataset (from torch.utils.data import Dataset)
as parent class?

**Reply by marvelousxmax** [reply]
![image](https://github.com/mrdbourke/pytorch-deep-learning/assets/131393386/82479623-6c05-4592-b154-e376d6f01251)

![image](https://github.com/mrdbourke/pytorch-deep-learning/assets/131393386/6b4e15c8-a404-4356-bdf4-c37337f76794)

**Reply by sametucuncu** [reply]
Your initialization function has typing error. It should be:
`def __init__(self, ... )`

**Reply by marvelousxmax** [reply]
Gotcha. I see the it now. Thank you for the second pair of eyes! I rewrote it and sorted the code. 

Now, I got this new error `TypeError: cannot unpack non-iterable NoneType object`. 

Does the line ( `self.classes, self.class_to_idx = find_classes(targ_dir)' ) needs to be created in a new list? 

![image](https://github.com/mrdbourke/pytorch-deep-learning/assets/131393386/0c81cdee-de5a-475d-b86b-208a1e9110b2)

**Reply by sametucuncu** [reply]
Could you check your train_dir definition?
